activemq-client/src/main/java/org/apache/activemq/management/LongStatistic.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.activemq.management;
+
+/**
+ * A statistic to store a single long value that is not incremented
+ * 
+ * Example: Store a timestamp value of a recent message
+ * 
+ */
+public interface LongStatistic extends UnsampledStatistic {
```

Comment:
You don't need these interfaces if you just use generics as I mentioned in my other comment.

Reply:
Yeah, I think this could be changed, but I'm wondering if it gets verbose to the consumer of the metric. I felt my original approach was "following the as-is" vs getting to the optimal target state.

If we go generic, do we add a generic parameter for all the dimensions? Sampled, Summed, Ranged, Fixed value, etc?

I think the interfaces do a good job of communicating the type. Feels like adding generics just pushes the type info to the consuming service code.

```
   public UnsampledStatisticImpl<Long> getDequeuedMessageBrokerOutTime() {
        return dequeuedMessageBrokerOutTime;
    }
```

I'm not married to either approach.

Reply:
I think it mostly depends on if the different types will have any logic that is different or needs to be overriden. You could also move common code into an abstract class if needed, I was just noticing a lot of copy/paste and seemed like generics could be useful.